src/app.rs

```diff
@@ -269,6 +282,7 @@ impl App {
             term_width: Term::stdout().size().1 as usize,
             files,
             theme: self.matches.value_of("theme"),
+            line_range: self.matches.value_of("line-range").map(LineRange::from),
```

Comment:
If we change the signature of `LineRange::from`, as suggested below, we could use
``` rust
line_range: tranpose(self.matches.value_of("line-range").map(LineRange::from))
```
such that the parsing-error-messages would bubble up to the main function (and reported to the user).

Here, `transpose` is a little helper function that turns a "Option of Result" into a "Result of Option". It could be implemented like this:
```
fn transpose<T, E>(opt: Option<Result<T, E>>) -> Result<Option<T>, E> {
    opt.map_or(Ok(None), |res| res.map(Some))
}
```

This function is actually an experimental feature which means that we could soon replace this by a call to the standard-library `.transpose()`: https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.transpose

Reply:
OK I think I see what you are saying now. We want to bubble up any errors that happen when parsing the linerange to bubble up from the config() function.